templates/common/_base/units/machine-config-daemon-firstboot.service.yaml

```diff
@@ -16,6 +16,8 @@ contents: |
   # Disable existing repos (if any) so that OS extensions would use embedded RPMs only
   ExecStartPre=-/usr/bin/sh -c "sed -i 's/enabled=1/enabled=0/' /etc/yum.repos.d/*.repo"
   ExecStart=/run/bin/machine-config-daemon firstboot-complete-machineconfig
+  Restart=on-failure
```

Comment:
This is not allowed for Type=oneshot service.
I build new image including this PR and deploy the cluster, machine-config-daemon-firstboot.service has a bad unit file setting.

[root@reliability01-8hnll-master-0 ~]# systemctl status machine-config-daemon-firstboot.service
● machine-config-daemon-firstboot.service - Machine Config Daemon Firstboot
   Loaded: bad-setting (Reason: Unit machine-config-daemon-firstboot.service has a bad unit file setting.)
   Active: inactive (dead)

Apr 08 02:13:14 reliability01-8hnll-master-0 systemd[1]: machine-config-daemon-firstboot.service: Service has Restart= setting other than no, which isn't allowed for Type=oneshot services. Refusing.

Reply:
I guess the other consideration is that, it is theoretically possible that subsequent runs fail in a different manner due to the original failure and mask over it.

Hmm, we very rarely fail on firstboot though. What timeout is being reached in the original BZ?

Reply:
> This is not allowed for Type=oneshot service.

Ah right.  Thanks for testing!  So I think the fix is to move the infinite retry loop into our Go code.

> I guess the other consideration is that, it is theoretically possible that subsequent runs fail in a different manner due to the original failure and mask over it.

We'd see all errors in the journal.